integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/SparkPropertyFacetBuilder.java

```diff
@@ -25,26 +26,19 @@ public class SparkPropertyFacetBuilder
       new HashSet<>(Arrays.asList("spark.master", "spark.app.name"));
   private static final String ALLOWED_PROPERTIES_KEY = "spark.openlineage.capturedProperties";
   private SparkConf conf;
-  private Set<String> allowerProperties;
+  private Set<String> allowedProperties;
 
   public SparkPropertyFacetBuilder(OpenLineageContext context) {
-    fillConfAndAllowerProperties(context.getSparkContext());
+    fillConfAndAllowedProperties(context.getSparkContext());
   }
 
   public SparkPropertyFacetBuilder() {
-    try {
-      SparkSession session = SparkSession.active();
-      fillConfAndAllowerProperties(session.sparkContext());
-    } catch (IllegalStateException ie) {
-      log.info("No active or default Spark session found");
-      conf = new SparkConf();
-      allowerProperties = new HashSet<>();
-    }
+    fillConfAndAllowedProperties(SparkSessionUtils.activeSession().map(ctx -> ctx.sparkContext()));
   }
 
-  private void fillConfAndAllowerProperties(SparkContext context) {
-    conf = context.getConf();
-    allowerProperties =
+  private void fillConfAndAllowedProperties(Optional<SparkContext> context) {
```

Comment:
Can we avoid `Optional` passed as method param? 
Perhaps the method SparkConf as param.